thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/TimeBasedAnomalyMerger.java

```diff
@@ -271,36 +271,60 @@ private void computeMergedAnomalyInfo(MergedAnomalyResultDTO mergedAnomalies, An
       anomalyFunction.updateMergedAnomalyInfo(mergedAnomalies, metricTimeSeries, windowStart, windowEnd, knownAnomalies);
 
       if(anomalyFunctionSpec.isToCalculateGlobalMetric()) {
-        computeImpactToGlobalMetric(adInputContext.getGlobalMetric(), mergedAnomalies);
+        MetricTimeSeries subMetricTimeSeries = adInputContext.getDimensionKeyMetricTimeSeriesMap().get(dimensions);
+        computeImpactToGlobalMetric(adInputContext.getGlobalMetric(), subMetricTimeSeries, mergedAnomalies);
       }
     }
   }
 
-  private void computeImpactToGlobalMetric(MetricTimeSeries globalMetricTimeSerise, MergedAnomalyResultDTO mergedAnomaly) {
+  /**
+   * Calculate the impact to global metric for the MergedAnomalyResult instance
+   *
+   * impact_to_global = severity * traffic_contribution
+   * trafic_contribution = sum(subMetric)/sum(globalMetric)
```

Comment:
You are doing average here given they may have different data points in the metric series countGlobal != countSub